app/models/embedded_ansible_worker/object_management.rb

```diff
@@ -25,20 +25,26 @@ def ensure_organization(provider, connection)
 
   def ensure_credential(provider, connection)
     return if provider.default_credential
+    ensure_organization(provider, connection) unless provider.default_organization
```

Comment:
Currently your `ensure_*` methods return an id if the thing was just created or `nil` if it already existed.  That feels a little strange and I don't think the return value is used much right now.
If however, you changed it to always return the id (already existing or just created), I think it might reduce some of the duplicate code.  For example, this could become something like:
```ruby
def ensure_credential(provider, connection)
  provider.default_credential ||= begin
    attrs = default_credential_attributes(ensure_organization)
    connection.api.credentials.create!(attrs).id
  end
end
```